scipy-stubs/interpolate/_bsplines.pyi

```diff
@@ -162,7 +162,7 @@ def make_lsq_spline(
     check_finite: onp.ToBool = True,
     *,
     method: _LSQMethod = "qr",
-) -> BSpline[np.complex128]: ...
+) -> BSpline[np.float64]: ...
```

Comment:
Yea you're right; if `y` is real, then the spline coefficients will also be real. However, that still leaves the overload above this one, which will *only* be used for complex `y`. In that, the returned spline coefs will be complex too. However, it is currently incorrectly annotated to return `BSpline[np.float64]` instead of `BSpline[np.complex128]`.

So instead of changing this return value to `float64`, it's probably easier to swap the annotation of `y` with the overload above. That way the float overload comes first, and the complex one second, which feels a bit more natural to me.